node-net-manager/env/EnvironmentManager.go

```diff
@@ -560,3 +602,194 @@ func (env *Environment) generateAddress() (net.IP, error) {
 func (env *Environment) freeContainerAddress(ip net.IP) {
 	env.addrCache = append(env.addrCache, ip)
 }
+
+/*
+	Unikernel
+*/
+
+func (env *Environment) CreateUnikernelNetwork(instancenumber int, name string, portmapping string) (net.IP, error) {
```

Comment:
I think it's alright.  But we should probably do what we did for the NodeEngine. Sort of a dispatcher that returns the implementation for the virtualization technology selected. Otherwise, we'll keep adding methods to this file for each virt technology. If you like the idea, this can either be solved here, or an issue can be opened, and this can be moved later.

Reply:
Would the new structure in commit d24c505 be better?